### PR TITLE
[ticket/14177] Uses Symfony's Debug ErrorHandler in development environment

### PR DESCRIPTION
**Pull request by Nicofuma**
PHPBB3-14177

### CODE REVIEW

File: phpBB/phpbb/debug/debug.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	 *
+	 * If the Symfony ClassLoader component is available, a special
+	 * class loader is also registered.
+	 *
+	 * @param int  $errorReportingLevel The level of error reporting you want
+	 * @param bool $displayErrors       Whether to display errors (for development) or just log them (for production)
+	 */
+	public static function enable($errorReportingLevel = null, $displayErrors = true)
+	{
+		if (static::$enabled) {
```
Comment (marc1706) [inline phpBB/phpbb/debug/debug.php]: Braces need to be on their own line according to our coding guidelines

File: phpBB/phpbb/debug/error_handler.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace phpbb\debug;
+
+use Symfony\Component\Debug\ErrorHandler;
+use Symfony\Component\Filesystem\Exception\IOException;
+
+class error_handler extends ErrorHandler
+{
+	public function handleError($type, $message, $file, $line, array $context, array $backtrace = null)
+	{
+		if ($type === E_USER_WARNING|| $type === E_USER_NOTICE)
```
Comment (marc1706) [inline phpBB/phpbb/debug/error_handler.php]: space missing after E_USER_WARNING

**Comment by Nicofuma** [comment]
updated. Should fix the CS issues

File: phpBB/phpbb/debug/debug.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		{
+			ini_set('display_errors', 0);
+			ExceptionHandler::register();
+		}
+		else if ($displayErrors && (!ini_get('log_errors') || ini_get('error_log')))
+		{
+			// CLI - display errors only if they're not already logged to STDERR
+			ini_set('display_errors', 1);
+		}
+		if ($displayErrors)
```
Comment (VSEphpbb) [inline phpBB/phpbb/debug/debug.php]: add an empty line above this if statement ?

Comment (Nicofuma) [inline phpBB/phpbb/debug/debug.php]: done

**Comment by marc1706** [comment]
:+1:

**Pull request merged**